src/NuGet.Core/NuGet.Protocol.Core.Types/UserAgentStringBuilder.cs

```diff
@@ -25,7 +24,9 @@ public UserAgentStringBuilder()
         public UserAgentStringBuilder(string clientName)
         {
             _clientName = clientName;
-            NuGetClientVersion = GetNuGetVersion();
+
+            // Read the client version from the assembly metadata and normalize it.
+            NuGetClientVersion = MinClientVersionUtility.GetNuGetClientVersion().ToNormalizedString();
```

Comment:
Calling `MinClientVersionUtility` here seems weird, since the minimum client version isn't really relevant here (AFAIK).